Mathlib/Data/List/Flatten.lean

```diff
@@ -34,121 +35,20 @@ protected theorem Sublist.flatMap {l₁ l₂ : List α} (h : l₁ <+ l₂) (f :
-  induction L generalizing i
-  · simp
-  · cases i <;> simp [take_append, *]
-
-@[deprecated (since := "2024-10-25")] alias take_sum_join' := take_sum_flatten'
-
-set_option linter.deprecated false in
-/-- In a join, dropping all the elements up to an index which is the sum of the lengths of the
-first `i` sublists, is the same as taking the join after dropping the first `i` sublists.
-
-See `List.drop_sum_flatten` for the corresponding statement using `List.sum`. -/
-@[deprecated "Use `List.drop_sum_flatten`." (since := "2024-10-17")]
-theorem drop_sum_flatten' (L : List (List α)) (i : ℕ) :
-    L.flatten.drop (Nat.sum ((L.map length).take i)) = (L.drop i).flatten := by
-  induction L generalizing i
-  · simp
-  · cases i <;> simp [drop_append, *]
-
-@[deprecated (since := "2024-10-25")] alias drop_sum_join' := drop_sum_flatten'
+@[deprecated (since := "2024-10-25")] alias length_join' := length_flatten
+@[deprecated (since := "2024-10-25")] alias countP_join' := countP_flatten
+@[deprecated (since := "2024-10-25")] alias count_join' := count_flatten
 
 /-- Taking only the first `i+1` elements in a list, and then dropping the first `i` ones, one is
 left with a list of length `1` made of the `i`-th element of the original list. -/
 theorem drop_take_succ_eq_cons_getElem (L : List α) (i : Nat) (h : i < L.length) :
     (L.take (i + 1)).drop i = [L[i]] := by
-  induction' L with head tail ih generalizing i
-  · exact (Nat.not_succ_le_zero i h).elim
-  rcases i with _ | i
-  · simp
-  · simpa using ih _ (by simpa using h)
-
-set_option linter.deprecated false in
-/-- In a flatten of sublists, taking the slice between the indices `A` and `B - 1` gives back the
-original sublist of index `i` if `A` is the sum of the lengths of sublists of index `< i`, and
-`B` is the sum of the lengths of sublists of index `≤ i`.
-
-See `List.drop_take_succ_flatten_eq_getElem` for the corresponding statement using `List.sum`. -/
-@[deprecated "Use `List.drop_take_succ_flatten_eq_getElem`." (since := "2024-10-17")]
-theorem drop_take_succ_flatten_eq_getElem' (L : List (List α)) (i : Nat) (h : i <  L.length) :
-    (L.flatten.take (Nat.sum ((L.map length).take (i + 1)))).drop
-      (Nat.sum ((L.map length).take i)) = L[i] := by
-  have : (L.map length).take i = ((L.take (i + 1)).map length).take i := by
-    simp [map_take, take_take, Nat.min_eq_left]
-  simp only [this, length_map, take_sum_flatten', drop_sum_flatten',
-    drop_take_succ_eq_cons_getElem, h, flatten, append_nil]
-
-@[deprecated (since := "2024-10-15")]
-alias drop_take_succ_join_eq_getElem' := drop_take_succ_flatten_eq_getElem'
-
-theorem flatten_drop_length_sub_one {L : List (List α)} (h : L ≠ []) :
-    (L.drop (L.length - 1)).flatten = L.getLast h := by
-  induction L using List.reverseRecOn
-  · cases h rfl
-  · simp
-
-@[deprecated (since := "2024-10-15")] alias join_drop_length_sub_one := flatten_drop_length_sub_one
+  induction L generalizing i with
```

Comment:
Please move this change to an independent PR, so we do not mix deleting deprecated declarations with proof rewriting.